writing-apps/our-first-app/continuous-integration.md

```diff
@@ -0,0 +1,49 @@
+# Continuous Integration
+
+Continuous integration testing (also called CI), is a way to automatically ensure that your app builds correctly and has the minimum necessary configuration to run when it's installed. Now that you have an app with a build system, metadata files, and packaging, we can configure CI using GitHub Actions.
+
+1. Navigate to your project's page on GitHub and select the "Actions" tab.
+
+2. Select "set up a workflow yourself". You'll then be shown the `main.yml` file with GitHub's default CI configuration. Replace the default workflow with the following:
+
+```yml
+name: CI
+
+# This workflow will run for any pull request or pushed commit
+on: [push, pull_request]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "flatpak"
+  flatpak:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # This job runs in a special container designed for building Flatpaks
+    container:
+      image: docker.io/bilelmoussaoui/flatpak-github-actions
```

Comment:
This should probably use a specific elementary image to avoid re-installing the SDK every time a build job is triggered, I will setup one on the action side unless you want to provide one yourself?

Reply:
That would be really appreciated if you wanted to set up a docker image!

Reply:
We already have a build job that's run weekly for building/updating the images, should be fairly easy to add a new one. I will let you know once it's there. What would be a proper tag? `elementary-X`? Note we have `gnome/freedesktop/kde-X` as well. In general X = the runtime version, but we can use something for elementary if you want

Reply:
Yeah I think that makes sense to use the runtime version. So for example, currently it would be `elementary-0.1.0`. I think that makes sense so people can easily keep it in sync